drivers/rtc/rtc_rv8263.c

```diff
@@ -0,0 +1,662 @@
+        err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_WEEKDAY_ALARM,
+                                    RV8263C8_BM_ALARM_ENABLE | timeptr->tm_wday);
+    } else {
+        err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_WEEKDAY_ALARM, RV8263C8_BM_ALARM_DISABLE);
+    }
+
+    if (err < 0) {
+        LOG_ERR("Error while writing WEEKDAY alarm! Error: %i", err);
+        return err;
+    }
+
+    return 0;
+}
+
+static int rv8263c8_alarm_get_time(const struct device *dev, uint16_t id, uint16_t *p_mask,
+                                   struct rtc_time *timeptr)
+{
+    int err;
+    const struct rv8263c8_config *config = dev->config;
+    uint8_t value[5];
+
+    if ((timeptr == NULL) || (id != 0)) {
+        return -EINVAL;
+    }
+
+    (*p_mask) = 0;
+
+    err = i2c_burst_read_dt(&config->i2c_bus, RV8263C8_REGISTER_SECONDS_ALARM, value, sizeof(value));
+    if (err < 0) {
+        LOG_ERR("Error while reading alarm! Error: %i", err);
+        return err;
+    }
+
+    if (value[0] <= MAX_SEC) {
+        timeptr->tm_sec = value[0];
+        (*p_mask) |= RTC_ALARM_TIME_MASK_SECOND;
+    }
+
+    if (value[1] <= MAX_MIN) {
+        timeptr->tm_min = value[1];
+        (*p_mask) |= RTC_ALARM_TIME_MASK_MINUTE;
+    }
+
+    if (value[2] <= MAX_HOUR) {
+        timeptr->tm_hour = value[2];
+        (*p_mask) |= RTC_ALARM_TIME_MASK_HOUR;
+    }
+
+    if (value[3] <= MAX_MDAY) {
+        timeptr->tm_mday = value[3];
+        (*p_mask) |= RTC_ALARM_TIME_MASK_MONTHDAY;
+    }
+
+    if (value[4] <= MAX_WDAY) {
+        timeptr->tm_hour = value[4];
+        (*p_mask) |= RTC_ALARM_TIME_MASK_WEEKDAY;
+    }
+
+    return 0;
+}
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+static int rv8263c8_alarm_set_callback(const struct device *dev, uint16_t id,
+                                       rtc_alarm_callback callback, void *p_user_data)
+{
+    int err;
+    struct rv8263c8_data *const data = dev->data;
+    const struct rv8263c8_config *config = dev->config;
+
+    if (id != 0) {
+        return -EINVAL;
+    }
+
+    (void)k_sem_take(&data->lock, K_FOREVER);
+    data->alarm_cb = callback;
+    data->alarm_cb_data = p_user_data;
+    k_sem_give(&data->lock);
+
+    if (callback != NULL) {
+        err = i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2, RV8263C8_BM_ALARM_INT_ENABLE,
+                                     RV8263C8_BM_ALARM_INT_ENABLE);
+    } else {
+        err = i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2, RV8263C8_BM_ALARM_INT_ENABLE,
+                                     RV8263C8_BM_ALARM_INT_DISABLE);
+    }
+
+    if (err < 0) {
+        LOG_ERR("Error while writing CONTROL2! Error: %i", err);
+        return err;
+    }
+
+    // When the callback is enabled the callback will be invoked and the alarm pending status bit will be cleared after it.
+    if (data->alarm_cb) {
+        data->alarm_cb(dev, id, data->alarm_cb_data);
```

Comment:
You should invoke the callback only if the alarm flag is already set.

Reply:
Yeah I´ve read the documentation wrong. I will remove it.